### PR TITLE
Release 1.0.0

### PR DESCRIPTION
**Pull request by itrujnara**
This is the first release of nf-core/reportho.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/reportho/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, also make a PR on the nf-core/reportho _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -27,8 +26,11 @@ If you're not used to this workflow with git, you can start with some [docs from
 
 ## Tests
 
-You can optionally test your changes by running the pipeline locally. Then it is recommended to use the `debug` profile to
-receive warnings about process selectors and other debug info. Example: `nextflow run . -profile debug,test,docker --outdir <OUTDIR>`.
+You have the option to test your changes locally by running the pipeline. For receiving warnings about process selectors and other `debug` information, it is recommended to use the debug profile. Execute all the tests with the following command:
+
+```bash
+nf-test test --profile debug,test,docker --verbose
```
Comment (mirpedrol) [inline .github/CONTRIBUTING.md]: The pipeline doesn't have pipeline-level nf-tests so it is better to keep the previous line `nextflow run . -profile debug,test,docker --outdir <OUTDIR>`

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,14 +3,64 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## v1.0dev - [date]
+## [v1.0.0](https://github.com/nf-core/reportho/releases/tag/1.0.0) - Magnificent Mainsail - [2024-06-06]
```
Comment (mirpedrol) [inline CHANGELOG.md]: Just a reminder to update the date before release :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,14 +3,64 @@
...
 
+The pipeline was created. In particular, it has the following features:
+
+- fetching of ortholog predictions from public databases, through APIs and from local snapshots
+- systematic comparison of the predictions and calculation of comparison statistics
+- creation of an ortholog list with user-defined criteria
+- basic downstream analysis of the obtained ortholog list
+- generation of a human-readable report
+
 ### `Fixed`
```
Comment (mirpedrol) [inline CHANGELOG.md]: The section could be removed if nothing was fixed

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,14 +3,64 @@
...
+At release date, the following database versions were current and used for testing the pipeline:
+
+| Database       | Version       |
+| -------------- | ------------- |
+| OMA            | Jul2023       |
+| PANTHER        | 18            |
+| OrthoInspector | Eukaryota2023 |
+| EggNOG         | 5.0           |
+
 ### `Deprecated`
```
Comment (mirpedrol) [inline CHANGELOG.md]: The section could be removed

File: conf/test_fasta.config [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Nextflow config file for running minimal tests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    Defines input files and everything required to run a fast and simple pipeline test.
+
+    Use as follows:
+        nextflow run nf-core/reportho -profile test,<docker/singularity> --outdir <OUTDIR>
```
Comment (mirpedrol) [inline conf/test_fasta.config]:
```suggestion
        nextflow run nf-core/reportho -profile test_fasta,<docker/singularity> --outdir <OUTDIR>
```

File: conf/test_fasta.config [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+----------------------------------------------------------------------------------------
+*/
+
+params {
+    config_profile_name        = 'Test profile'
```
Comment (mirpedrol) [inline conf/test_fasta.config]:
```suggestion
    config_profile_name        = 'Test profile with fasta'
```
or any other name that you like but can differentiate it from `test.config` :)

File: conf/test_offline.config [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    Defines input files and everything required to run a fast and simple pipeline test.
+
+    Use as follows:
+        nextflow run nf-core/reportho -profile test_offline,<docker/singularity> --outdir <OUTDIR>
+
+----------------------------------------------------------------------------------------
+*/
+
+params {
+    config_profile_name        = 'Test profile'
```
Comment (mirpedrol) [inline conf/test_offline.config]:
```suggestion
    config_profile_name        = 'Test profile offline'
```

File: docs/images/nf-core-reportho_tube_map_beta.png
Comment (ramprasadn) [inline docs/images/nf-core-reportho_tube_map_beta.png:1]: Is this file still necessary?

File: docs/images/reportho_tube_map.svg
Comment (ramprasadn) [inline docs/images/reportho_tube_map.svg:1]: Perhaps the map should say version 1.0.0 and not dev? ;)

File: modules/local/dump_params.nf [hunk 8]
```diff
@@ -0,0 +1,53 @@
+process DUMP_PARAMS {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
```
Comment (ramprasadn) [inline modules/local/dump_params.nf:6]: conda definition is missing here

File: modules/local/fetch_panther_group_online.nf [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    conda "conda-forge::python=3.11.0 conda-forge::biopython=1.83.0 conda-forge::requests=2.31.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-bc54124b36864a4af42a9db48b90a404b5869e7e:5258b8e5ba20587b7cbf3e942e973af5045a1e59-0' :
+        'biocontainers/mulled-v2-bc54124b36864a4af42a9db48b90a404b5869e7e:5258b8e5ba20587b7cbf3e942e973af5045a1e59-0' }"
+
+    input:
+    tuple val(meta), path(uniprot_id), path(taxid), path(exact)
+
+    output:
+    tuple val(meta), path("*_panther_group.csv"), emit:panther_group
```
Comment (mirpedrol) [inline modules/local/fetch_panther_group_online.nf]:
```suggestion
    tuple val(meta), path("*_panther_group.csv"), emit: panther_group
```

File: modules/local/plot_orthologs.nf [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        'community.wave.seqera.io/library/r-ggvenndiagram_r-reshape2_r-tidyverse:6ab82708ae578c26' }"
+
+    input:
+    tuple val(meta), path(score_table)
+
+    output:
+    tuple val(meta), path("*_supports_light.png"), path("*_supports_dark.png"), emit: supports
+    tuple val(meta), path("*_venn_light.png"), path("*_venn_dark.png")        , emit: venn
+    tuple val(meta), path("*_jaccard_light.png"), path("*_jaccard_dark.png")  , emit: jaccard
+    path "versions.yml"                          , emit: versions
```
Comment (mirpedrol) [inline modules/local/plot_orthologs.nf]: could align the emit

File: modules/local/plot_tree.nf [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        'oras://community.wave.seqera.io/library/bioconductor-ggtree_bioconductor-treeio_r-ggplot2:89a30ee47c501fe4' :
+        'community.wave.seqera.io/library/bioconductor-ggtree_bioconductor-treeio_r-ggplot2:54fc04b8b0f7b6c7' }"
+
+    input:
+    tuple val(meta), path(tree)
+    val method
+
+    output:
+    tuple val(meta), path("*_light.png"), path("*_dark.png") , emit: plot
+    path "versions.yml"                 , emit: versions
```
Comment (mirpedrol) [inline modules/local/plot_tree.nf]: align

File: nextflow_schema.json [hunk 12]
```diff
@@ -43,34 +49,229 @@
...
+                    "type": "boolean",
+                    "default": "false",
+                    "description": "Use all ortholog search methods. Will mix online and local methods if needed. Overrides all individual database flags.",
+                    "help_text": "If set to `true`, the pipeline will use all ortholog search methods.",
+                    "fa_icon": "fas fa-database"
+                },
+                "local_databases": {
+                    "type": "boolean",
+                    "default": "false",
+                    "description": "Use local databases for the analysis. If use_all is set to `true`, online databases might still be used.",
```
Comment (mirpedrol) [inline nextflow_schema.json]: > If use_all is set to `true`, online databases might still be used.

Is this true? I think you give `--local_databases` preference. Otherwise, the error in `subworkflows/local/utils_nfcore_reportho_pipeline/main.nf` must be changed.

File: subworkflows/local/fetch_sequences.nf
Comment (ramprasadn) [inline subworkflows/local/fetch_sequences.nf:1]: Do you think it is necessary for this step to have its own subworkflow? Considering it only involves one module, I suggest not creating a separate subworkflow for it.

File: subworkflows/local/fetch_structures.nf
Comment (ramprasadn) [inline subworkflows/local/fetch_structures.nf:1]: This one too

Comment (mirpedrol) [inline subworkflows/local/fetch_structures.nf:1]: Is this subworkflow needed? it is only running one process

File: subworkflows/local/get_orthologs.nf [hunk 13]
```diff
@@ -0,0 +1,274 @@
...
+
+    ch_samplesheet_fasta
+        .map { it -> [it[0], file(it[1])] }
+        .set { ch_fasta }
+
+    IDENTIFY_SEQ_ONLINE (
+        ch_fasta
+    )
+
+    ch_query = IDENTIFY_SEQ_ONLINE.out.seqinfo
```
Comment (mirpedrol) [inline subworkflows/local/get_orthologs.nf]: Not needed. `ch_query` is overwritten later

File: subworkflows/local/get_orthologs.nf [hunk 14]
```diff
@@ -0,0 +1,274 @@
...
+    ch_versions     = Channel.empty()
+    ch_orthogroups  = Channel.empty()
+
+    ch_oma_groups   = params.oma_path ? Channel.value(file(params.oma_path)) : Channel.empty()
+    ch_oma_uniprot  = params.oma_uniprot_path ? Channel.value(file(params.oma_uniprot_path)) : Channel.empty()
+    ch_oma_ensembl  = params.oma_ensembl_path ? Channel.value(file(params.oma_ensembl_path)) : Channel.empty()
+    ch_oma_refseq   = params.oma_refseq_path ? Channel.value(file(params.oma_refseq_path)) : Channel.empty()
+    ch_panther      = params.panther_path ? Channel.value(file(params.panther_path)) : Channel.empty()
+    ch_eggnog       = params.eggnog_path ? Channel.value(file(params.eggnog_path)) : Channel.empty()
+    ch_eggnog_idmap = params.eggnog_idmap_path ? Channel.value(file(params.eggnog_idmap_path)) : Channel.empty()
```
Comment (ramprasadn) [inline subworkflows/local/get_orthologs.nf]: I also suggest moving these statements to reportho.nf, as that is where other nf-core pipelines usually perform channel initializations from parameters. One of the benefits of doing so is that it enables people to easily share subworkflows, and personally, it seems more logical to handle this there.

**Review by mirpedrol — COMMENTED** [review]
LGTM! only a couple of comments

**Review by ramprasadn — COMMENTED** [review]
Nice work @itrujnara! A few minor comments

File: nextflow_schema.json (hunk 12 above)
Comment (itrujnara) [reply nextflow_schema.json]: This is not true anymore. Forgot to change this description.

**Review by ramprasadn — APPROVED** [review]
LGTM! 🚀

**Review by mirpedrol — APPROVED** [review]
LGTM!
